src/OpenStack.Net/OpenStack/Services/ObjectStorage/V1/StorageMetadata.cs

```diff
@@ -0,0 +1,255 @@
+﻿namespace OpenStack.Services.ObjectStorage.V1
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Collections.Immutable;
+    using System.Linq;
+    using System.Net.Http;
+    using System.Text;
+    using OpenStack.Collections;
+
+    /// <summary>
+    /// This is the base class for representing metadata associated with resources in the Object Storage Service.
+    /// </summary>
+    /// <threadsafety static="true" instance="false"/>
```

Comment:
The `StorageMetadata` types (this and derived) are immutable. Both static and instance members are thread-safe.